### PR TITLE
Feature 2309 team skills report query parameters

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/member_selector/MemberSelector.jsx [hunk 1]
```diff
@@ -80,14 +80,9 @@ const MemberSelector = ({
   className,
   style
 }) => {
-  const isControlled = !!selected && Array.isArray(selected);
```
Comment (mvolkmann) [inline web-ui/src/components/member_selector/MemberSelector.jsx:83]: Previously this component maintained its own state of selected members in addition to allowing them to be passed in by a component that also maintains that same state. I changed this to require the selected members to be passed in using the `selected` prop. The function specified by the `onChange` prop is called every time the selection of members changes so the parent component can update its state. This simplifies the code and works much better with our custom hook `useQueryParameters`.

File: web-ui/src/pages/TeamSkillReportPage.jsx [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import React, { useContext, useState } from 'react';
+import React, { useContext, useRef, useState } from 'react';
+
+import { Autocomplete, Button, TextField, Typography } from '@mui/material';
```
Comment (mvolkmann) [inline web-ui/src/pages/TeamSkillReportPage.jsx:3]: I combined some MUI imports.

File: web-ui/src/pages/TeamSkillReportPage.jsx [hunk 3]
```diff
@@ -129,7 +163,8 @@ const TeamSkillReportPage = () => {
       <MemberSelector
         className="team-skill-member-selector"
         listHeight={300}
-        onChange={selected => setSelectedMembers(selected)}
+        onChange={setSelectedMembers}
+        selected={selectedMembers}
```
Comment (mvolkmann) [inline web-ui/src/pages/TeamSkillReportPage.jsx:167]: The `MemberSelector` component now requires the `selected` prop.

**Review by S78901 — APPROVED** [review]

**Review by vhscom — APPROVED** [review]

File: web-ui/src/components/member_selector/MemberSelector.jsx (hunk 1 above)
Comment (vhscom) [reply web-ui/src/components/member_selector/MemberSelector.jsx:83]: This component is a little tricky I'm noticing, as both the earlier and the current implementations do not seem to update the member selection when the `selected` prop changes (e.g. during runtime as data becomes available from the server) in the controlling parent as I'd expected. Typically with controlled components, changes to the _controlling_ props (like `selected` in this case) should cause a re-render of the child. That's outside the scope here but worth noting.

Comment (vhscom) [reply web-ui/src/components/member_selector/MemberSelector.jsx:83]: Was a little tricky. Now it works like a dream. ✨

**Review by S78901 — APPROVED** [review]

**Pull request merged**